include/eosio/vm/types.hpp

```diff
@@ -189,11 +189,140 @@ namespace eosio { namespace vm {
+         uint32_t get_imported_functions_size() const {
+            return get_imported_functions_size_impl(imports);
+         }
+      };
+
+      // The memory storing module data for JIT
+      std::unique_ptr<jit_mod_t> jit_mod;
+
+      // Constructs data for JIT execution.
+      // Called from backend::construct() after parsing is finalized.
+      void make_jit_module() {
+         jit_mod = std::make_unique<jit_mod_t>();
+
+         for (uint32_t i = 0; i < types.size(); ++i) {
+            const auto& type = types[i];
+            jit_mod_t::jit_func_type func_type {
+               type.form,
+               {type.param_types.data(), type.param_types.data() + type.param_types.size()},
+               type.return_count,
+               type.return_type
+            };
+            jit_mod->types.emplace_back(func_type);
+         }
+
+         for (uint32_t i = 0; i < imports.size(); ++i) {
+            const auto& entry = imports[i];
+            jit_mod_t::jit_import_entry import_entry {
+               {entry.module_str.data(), entry.module_str.data() + entry.module_str.size()},
+               {entry.field_str.data(), entry.field_str.data() + entry.field_str.size()},
+               entry.kind,
+               {entry.type.func_t}
+            };
+            jit_mod->imports.emplace_back(import_entry);
+         }
+
+         if (memories.size() > 0) {
+            jit_mod->memories.emplace_back(memories[0]); // memories has one element only
+         }
+
+         if (functions.size() > 0) {
+            jit_mod->functions.assign(functions.data(), functions.data() + functions.size());
+         }
+
+         if (globals.size() > 0) {
+            jit_mod->globals.assign(globals.raw(), globals.raw() + globals.size());
+         }
+
+         for (uint32_t i = 0; i < exports.size(); ++i) {
+            const auto& entry = exports[i];
+            jit_mod_t::jit_export_entry export_entry {
+               {entry.field_str.data(), entry.field_str.data() + entry.field_str.size()},
+               entry.kind,
+               entry.index
+            };
+            jit_mod->exports.emplace_back(export_entry);
+         }
+
+         for (uint32_t i = 0; i < code.size(); ++i) {
+            jit_mod->code_offset.emplace_back(code[i].jit_code_offset);
+         }
+
+         for (uint32_t i = 0; i < data.size(); ++i) {
+            const auto& data_seg = data[i];
+            jit_mod_t::jit_data_segment jit_mod_seg{
+               data_seg.index,
+               data_seg.offset,
+               {data_seg.data.data(), data_seg.data.data() + data_seg.data.size()}
+            };
+            jit_mod->data.emplace_back(jit_mod_seg);
+         }
+
+         if (import_functions.size() > 0) {
+            jit_mod->import_functions.assign(import_functions.data(), import_functions.data() + import_functions.size());
+         }
+
+         // Important. Release the memory used during parsing.
+         allocator.release_base_memory();
```

Comment:
This seems a bit out of place to me. I think the caller should release memory after calling `make_jit_module()`. I didn't expect `make_jit_module()` to have a side-effect of releasing the base memory.

Reply:
Thanks! I was debating if here was the right place to release the memory, even considering changing the method name to reflect it. Will move the release out to the caller.